### PR TITLE
feat(query):  parallel read of ndjson in copy.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

`split_size` is 0 by default, which means splitting is disabled.

part of https://github.com/datafuselabs/databend/issues/7732

### CODE REVIEW

File: src/meta/proto-conv/tests/it/user_proto_conv.rs [hunk 1]
```diff
@@ -156,6 +158,7 @@ pub(crate) fn test_s3_stage_info_v14() -> mt::UserStageInfo {
         copy_options: mt::CopyOptions {
             on_error: mt::OnErrorMode::SkipFileNum(666),
             size_limit: 1038,
+            split_size: 0,
```
Comment (ClSlaid) [inline src/meta/proto-conv/tests/it/user_proto_conv.rs:161]: Please update the version note vector [here](https://github.com/datafuselabs/databend/blob/1852734c653d7d3090cb4efce9ed2e35b8d5cc83/src/meta/proto-conv/src/util.rs#L64), and add new version of tests to affected ProtoBuf structs.

File: src/meta/proto-conv/tests/it/user_proto_conv.rs [hunk 2]
```diff
@@ -120,6 +121,7 @@ pub(crate) fn test_s3_stage_info() -> mt::UserStageInfo {
...
```
Comment (ClSlaid) [inline src/meta/proto-conv/tests/it/user_proto_conv.rs:124]: add a test_s3_stage_info_v16(), test_gcs_stage_info_v16() and test_oss_stage_info_v16() for new changes.

File: src/meta/proto-conv/tests/it/user_stage.rs [hunk 3]
```diff
@@ -91,6 +91,7 @@ fn test_user_stage_oss_v13() -> anyhow::Result<()> {
         copy_options: mt::CopyOptions {
             on_error: mt::OnErrorMode::SkipFileNum(666),
             size_limit: 1038,
+            split_size: 0,
```
Comment (ClSlaid) [inline src/meta/proto-conv/tests/it/user_stage.rs:94]: Add test_user_stage_s3_v16() test_user_stage_gcs_v16() and test_user_stage_oss_v16()

Comment (youngsofun) [reply src/meta/proto-conv/tests/it/user_stage.rs:292]: done

**Review by drmingdrmer — APPROVED** [review]

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**